pression/data/CompressorInfo.h

```diff
@@ -0,0 +1,52 @@
+
+/* Copyright (c) 2016, Stefan.Eilemann@epfl.ch
+ *
+ * This file is part of Pression <https://github.com/Eyescale/Pression>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#pragma once
+
+#include <pression/data/api.h>
+#include <pression/data/types.h>
+#include <functional>
+
+namespace pression
+{
+namespace data
+{
+struct CompressorInfo
+{
+    CompressorInfo()
+        : ratio( 1.f )
+        , speed( 1.f )
+        , create( []{ return nullptr; })
```

Comment:
Maybe it's better not initializing this function object. Calling a default constructed std::function throws an exception, whereas with the lambda function you're giving the caller needs to check for the return value.